core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftTransportKey.java

```diff
@@ -81,7 +92,7 @@ public boolean equals(Object o) {
       return false;
     }
     ThriftTransportKey ttk = (ThriftTransportKey) o;
-    return server.equals(ttk.server) && timeout == ttk.timeout
+    return type.equals(ttk.type) && server.equals(ttk.server) && timeout == ttk.timeout
```

Comment:
Looked at ThriftClientTypes and did not see that it implements equals() or hashCode().  Seems like the intention is that there is only one instance of type in the JVM so its ok to fall back to the object hashCode.  This makes it tricky to reason about the correctness, not sure of a way to improve this.  If possible it would be good to reduce/localize the amount of assumptions being made around this.  Thought through the following possible solutions, but they seemed like dead ends.

 * could we make ThriftClientTypes an enum?  Does not seem like this would work with the type parameter.
 * could we make ThriftClientTypes constructor private?  This makes it easier to reason about single instances.  The subclasses would probably need to be pulled in as inner classes to do this.  Not sure if its workable even then though.
 * could we add equals and hashCode impls that only look at the service name?  This feels dubious because it excludes the factory

This issue does not need to hold up this PR if the current code does only create a single instance per type in the JVM.  Did not attempt to verify if this is the case or not.

Reply:
> could we make ThriftClientTypes an enum? Does not seem like this would work with the type parameter.

IIRC I tried this when I originally created ThriftClientTypes and it didn't work with the generics

> could we make ThriftClientTypes constructor private? This makes it easier to reason about single instances. The subclasses would probably need to be pulled in as inner classes to do this. Not sure if its workable even then though.

I think we could probably make it `protected` vs `public`, or remove the modifier and make it package protected. I looked at all of the impls and their constructors have no modifier, so they are package protected.

Reply:
I could not make ThriftClientTypes constructor package protected. The ThriftProcessorTypes class in server-base needs access. I was able to make it `protected` however in 1a3d5e0